indicatorStatusIcon.js

```diff
@@ -65,6 +66,9 @@ class AppIndicators_IndicatorStatusIcon extends PanelMenu.Button {
 
         if (this._indicator.isReady)
             this._display()
+
+        this._settings = Convenience.getSettings();
+        Util.connectSmart(this._settings, 'changed::tray-pos', this, '_display');
```

Comment:
Use `this._display` as callback, the string versions aren't really easy to maintain

Reply:
Fixed. See comment on connectSmart further up.

Reply:
My fix did not work. Do not know how to proceed.